frontend/src/context/AuthContext.js

```diff
@@ -0,0 +1,70 @@
+import { createContext, useContext, useState } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { jwtDecode } from 'jwt-decode'
+
+const API_URL = process.env.REACT_APP_API_URL
+
+const AuthContext = createContext()
+
+
+export const AuthProvider = ({children}) => {
+  const navigate = useNavigate()
+  const [token, setToken] = useState(localStorage.getItem('accessToken') || null)
+  const [userId, setUserId] = useState(null)
+  const [role, setRole] = useState(null)
+
+  const checkUserRole = async (accessToken) => {
+    const decodedToken = jwtDecode(accessToken)
+    const { user_id} = decodedToken
+    setUserId(user_id)
+
+    try {
+      const res = await fetch(`${API_URL}/api/shelters/${user_id}`, {
+        headers: {
+          Authorization: `Bearer ${accessToken}`
+        }
+      })
+
+      if (!res.ok) return setRole('seeker')
```

Comment:
I get that this will set a role, but it doesn't keep the state on redirect or refresh, this should be stored in localstorage as well as in your navbar it can be null whe it checks for the role when the user is logged in. For example try logging in and refreshing the page

Reply:
Fixed in [f6a47fc](https://github.com/lujorryn/PetPal-G0346/pull/22/commits/f6a47fc78978550def9d1e1674e9d96960f35e95)